src/bounce_solution/action_calculation.cpp

```diff
@@ -474,7 +474,14 @@ std::vector<double> BounceActionInt::ExactSolutionLin(double l0,
       ss << rho_up << "\t" << l << "\t" << dVdl << "\t" << d2Vdl2 << "\n";
       BSMPT::Logger::Write(BSMPT::LoggingLevel::BounceDetailed, ss.str());
       ss.str(std::string());
-      return (ExactSolutionLin(l0, l / 10., dVdl, d2Vdl2));
+      if (l0 == l0 + (l - l0) / 10.)
```

Comment:
Isn't this check only true if `l0== 1` ?

Reply:
No, its true if ```l = l0```which is only true if ```(l - l0) / 10.``` is so small compared with ```l0```that gets truncated in the sum.
This checks if the recursive function arrived at the numerical precision limit.

Reply:
Ah, this is an `l` and not a 1.
Maybe it is clearer here to use something like `abs(l-l0) <= eps`?

Reply:
Done, I compared the difference with the total length. I also added a check that will prevent most of these issues.